fvm/environment/uuids.go

```diff
@@ -39,45 +50,118 @@ func (generator ParseRestrictedUUIDGenerator) GenerateUUID() (uint64, error) {
 
 type uUIDGenerator struct {
 	tracer tracing.TracerSpan
+	log    zerolog.Logger
 	meter  Meter
 
 	txnState state.NestedTransactionPreparer
+
+	blockHeader *flow.Header
+	txnIndex    uint32
```

Comment:
Is it correct? Or it's the index within a collection?
```suggestion
	txnIndex    uint32 // the transaction index within a block
```